rust-runtime/aws-smithy-cbor/src/decode.rs

```diff
@@ -223,8 +230,14 @@ impl<'b> Decoder<'b> {
                 "expected timestamp tag",
             )))
         } else {
+            // Values that are more granular than millisecond precision SHOULD be truncated to fit
+            // millisecond precision for epoch-seconds:
+            // https://smithy.io/2.0/spec/protocol-traits.html#timestamp-formats
             let epoch_seconds = self.decoder.f64().map_err(DeserializeError::new)?;
-            Ok(DateTime::from_secs_f64(epoch_seconds))
+            let mut result = DateTime::from_secs_f64(epoch_seconds);
+            let subsec_nanos = result.subsec_nanos();
+            result.set_subsec_nanos(subsec_nanos / 1_000_000 * 1_000_000);
```

Comment:
Isn't this always 0? `subsec_nanos` is a `u32`. `u32::MAX` is smaller than `10^6 * 10^6`.

Also, I would add a Rust unit test.

Reply:
`subsec_nanos / 1_000_000` is performed first and then multiplication. But added parentheses for clarify and added unit test in 4d0b0b5.